proposed/http-message-meta.md

```diff
@@ -613,6 +613,75 @@ Examples of this practice already exist in libraries such as
 has functionality for casting the value to a string, these objects can be
 used to populate the headers of an HTTP message.
 
+### What about handling arrays of file uploads?
+
+When submitting a form with file inputs named `file[0]` and `file[1]`, the
+contents of `$_FILES` becomes:
+
+```php
+array( 
+    'file' => array(
+        'name' => array(
+            0 => 'file0.txt',
+            1 => 'file1.txt',
+        ) ,
+        'type' => array(
+            0 => 'text/plain',
+            1 => 'text/html',
+        ),
+        /* and so on */
+    ), 
+)
+```
+
+This means that in such cases, you cannot simply access the file details for
+`file[1]` using the following:
+
+```php
+$fileDetails = $request->getFileParams()['file'][1];
+```
+
+as each specific file detail (e.g., content type) is nested with those of any
+sibling uploads.
+
+We considered specifying that `getFileParams()` MUST parse the values into a
+nested structure that mimics how they were provided in the form; however, we
+feel:
+
+- It introduces inconsistency in the API. Other parameters, such as query string
+  and cookie parameters, are specified such that they MUST retain the
+  structure of their corresponding superglobals. (Body parameters are a
```

Comment:
In ZF, we don't handle it at the request level, though. We handle it in our `Zend\File\Transfer ` component, and the related filters and validators. Those are fed the raw `$_FILES` structure, which the request object encapsulates. (I've actually been blissfully unaware of how we do this until investigating for this issue.)

If anything, that validates the approach suggested here.